lib/client.js

```diff
@@ -2,11 +2,13 @@
 
 import { Response } from './response';
 
+const $request = Symbol('request');
+
 export class Client {
-  constructor(config={}) {
-    // config.url
-    // config.token
-    // todo: set up properties
+  constructor({ url, token, request } = {}) {
+    assertFunction(request);
+
+    this[$request] = request;
```

Comment:
I was running through this sort of pattern when thinking through the Promises the other night. The only issue with this would be if methods in the other classes need to access it. For instance methods we can pass the client on construct, but then we would have to pass it again for any static methods which is kinda clunky, hence my comment (on slack) regarding a possible singleton object. Would be interested in hearing your thoughts on this, maybe I've got the wrong idea.